### PR TITLE
Remove types from GeoShapeQueryBuilder

### PR DESCRIPTION
**Pull request by romseygeek**
This commit removes the unused 'shapeType' information from `AbstractGeoQueryBuilder` 
and its implementations.

Related to #41059

### CODE REVIEW

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Comment by romseygeek** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

File: server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java [hunk 1]
```diff
@@ -86,27 +85,8 @@ public GeoShapeQueryBuilder(String fieldName, ShapeBuilder shape) {
...
-     * type
-     *
-     * @param fieldName
-     *            Name of the field that will be filtered
-     * @param indexedShapeId
-     *            ID of the indexed Shape that will be used in the Query
-     * @param indexedShapeType
-     *            Index type of the indexed Shapes
-     * @deprecated use {@link #GeoShapeQueryBuilder(String, String)} instead
-     */
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java]: Maybe keep and adapt the docs? Or do you think its not useful or self-explanatory?

**Review by cbuescher — APPROVED** [review]
LGTM, left a comment regarding leaving/modifying javadoc. Change and leave or ignore, I'm fine either way.

**Pull request merged**